Formula/newrelic-infra-agent.rb

```diff
@@ -0,0 +1,42 @@
+class NewrelicInfraAgent < Formula
+  desc "New Relic infrastructure agent"
+  homepage "https://github.com/newrelic/infrastructure-agent"
+  url "https://github.com/newrelic/infrastructure-agent/archive/refs/tags/1.20.3.tar.gz"
+  sha256 "3a29b19541e6f32fb87196a7833d08a9aed41b8ff62ef3a72326236d262cb033"
+  license "Apache-2.0"
+  head "https://github.com/newrelic/infrastructure-agent.git"
+
+  depends_on "go" => :build
+
+  def install
+    goarch = Hardware::CPU.arm? ? "arm64" : "amd64"
+    cgo = Hardware::CPU.arm? ? "0" : "1"
+    ENV["VERSION"] = "1.20.3"
+    ENV["GOOS"] = "darwin"
+    ENV["GOARCH"] = goarch
+    ENV["CGO_ENABLED"] = cgo
+    system "make", "dist-for-os"
+    on_macos do
+      bin.install "dist/darwin-newrelic-infra_darwin_#{goarch}/newrelic-infra"
+      bin.install "dist/darwin-newrelic-infra-ctl_darwin_#{goarch}/newrelic-infra-ctl"
+      bin.install "dist/darwin-newrelic-infra-service_darwin_#{goarch}/newrelic-infra-service"
+    end
+
+    (etc/"newrelic-infra").mkpath
+    (var/"log/newrelic-infra").mkpath
+    (var/"db/newrelic-infra").mkpath
```

Comment:
Move this into a `post_install` method. If you leave these in `install`, they'll just get cleaned up because they're empty.

Reply:
Can we reach archive internals in post install part? To be able to do this:
```
(var/"db/newrelic-infra").install "assets/licence/LICENSE.macos.txt"
```

Reply:
No. That part you can keep in `install`, and there's no need to do `mkpath` with `var/"newrelic/db"`.